File: week06/pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letter_count = count_letters(text)
    word_count = count_words(text)
    sentence_count = count_sentences(text)

    L = 100 * letter_count / word_count
    S = 100 * sentence_count / word_count

    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    counter = 0

    for c in text:
        if c.isalpha():
            counter += 1

    return counter


def count_words(text):
    counter = 0

    for c in text:
        if c.isspace():
            counter += 1

    # count last word in the text
    counter += 1

    return counter


def count_sentences(text):
    counter = 0

    for c in text:
        if c in ['.', '!', '?']:
            counter += 1

    return counter


if __name__ == "__main__":
    main()

File: week02/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letterCount = count_letters(text);
    int wordCount = count_words(text);
    int sentenceCount = count_sentences(text);

    float L = 100.0 * letterCount / wordCount;
    float S = 100.0 * sentenceCount / wordCount;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

int count_letters(string text)
{
    int counter = 0;
    int len = strlen(text);

    for (int i = 0; i < len; ++i)
    {
        if (isalpha(text[i]))
        {
            ++counter;
        }
    }

    return counter;
}

int count_words(string text)
{
    int counter = 0;
    int len = strlen(text);

    for (int i = 0; i < len; ++i)
    {
        if (isspace(text[i]))
        {
            ++counter;
        }
    }

    // Count last word in the sentence
    ++counter;

    return counter;
}

int count_sentences(string text)
{
    int counter = 0;
    int len = strlen(text);

    for (int i = 0; i < len; ++i)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            ++counter;
        }
    }

    return counter;
}
